tile/platform/local_machine/mem_cache.cc

```diff
@@ -8,6 +8,25 @@ namespace vertexai {
 namespace tile {
 namespace local_machine {
 
+MemCache::MemCache(const std::shared_ptr<DevInfo>& dev_info, hal::Memory* source) :
+  total_{0}, source_{source} {
+  mem_limit_ = dev_info->settings.max_global_mem();
+}
+
+void MemCache::Cleanup(size_t new_size) {
+  while (total_ + new_size > mem_limit_) {
+    // Out of device memory, clean up cache
+    while (mem_.size() > 0 && mem_.begin()->second.empty()) {
```

Comment:
What's the goal here?  The thing is, as soon as the current program finishes, it's going to release all of its temporary memory to this cache, which is just going to have to reallocate these table entries, right?

(If this is a good thing to do, why not do it directly at the points where memory is being allocated?)